### PR TITLE
Next gen twiml

### PR DESCRIPTION
**Pull request by jingming**

### CODE REVIEW

File: twilio/twiml/__init__.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            el.text = self.body
+
+        for verb in self.verbs:
+            el.append(verb.xml())
+
+        return el
+
+    def append(self, verb):
+        self.verbs.append(verb)
+        return verb
```
Comment (dougblack) [inline twilio/twiml/__init__.py]: Might be nicer to return `self` here.


```
response.append(Dial()).append(Sms())
```

File: twilio/twiml/__init__.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                                     True
+        """
+        xml = ET.tostring(self.xml()).decode('utf-8')
+
+        if xml_declaration:
+            return '<?xml version="1.0" encoding="UTF-8"?>' + xml
+        else:
+            return xml
+
+    def xml(self):
```
Comment (efossier) [inline twilio/twiml/__init__.py]: Docs?

File: twilio/twiml/messaging_response.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        self.name = 'Response'
+
+    def message(self,
+                body,
+                to=None,
+                from_=None,
+                method=None,
+                action=None,
+                status_callback=None,
+                **kwargs):
```
Comment (efossier) [inline twilio/twiml/messaging_response.py]: ...ew one line per param?

File: twilio/twiml/__init__.py (hunk 2 above)
Comment (efossier) [inline twilio/twiml/__init__.py]: or prefix with _ if not public

File: twilio/twiml/__init__.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        if self.body:
+            el.text = self.body
+
+        for verb in self.verbs:
+            el.append(verb.xml())
+
+        return el
+
+    def append(self, verb):
```
Comment (efossier) [inline twilio/twiml/__init__.py]: Check if its an instance of `TwiML`?

File: twilio/twiml/messaging_response.py (hunk 3 above)
Comment (dougblack) [inline twilio/twiml/messaging_response.py]: This is fine and valid.

File: twilio/twiml/__init__.py (hunk 4 above)
Comment (dougblack) [inline twilio/twiml/__init__.py]: I like this, it would help prevent mistakes.

File: twilio/twiml/__init__.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            return xml
+
+    def append(self, verb):
+        """
+        Add a TwiML doc
+        :param verb: TwiML Document
+        :return:
+        """
+        if not isinstance(verb, TwiML):
+            raise TwiMLException()
```
Comment (dougblack) [inline twilio/twiml/__init__.py]: Let's add a message like "can't nest responses".

File: twilio/twiml/messaging_response.py (hunk 3 above)
Comment (efossier) [inline twilio/twiml/messaging_response.py]: I like the other style where you keep them inline until the length limit then wrap to the next line. but not a big deal.

**Pull request merged**